core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -94,12 +91,14 @@ private[spark] abstract class YarnSchedulerBackend(
   /**
    * An actor that communicates with the ApplicationMaster.
    */
-  private class YarnSchedulerActor extends Actor {
+  protected class YarnSchedulerActor(isClusterMode: Boolean)  extends Actor {
```

Comment:
this seems like an inappropriate use of class hierarchy. Instead I would make this an abstract class, and declare a `YarnClientSchedulerActor` in `YarnClientSchedulerBackend` that overrides `preStart()`